src/ucp/rndv/rndv_put.c

```diff
@@ -514,6 +519,7 @@ ucp_proto_rndv_put_mtype_copy_progress(uct_pending_req_t *uct_req)
     req->flags |= UCP_REQUEST_FLAG_PROTO_INITIALIZED;
     ucp_proto_rndv_mtype_copy(req, req->send.rndv.mdesc->ptr,
                               ucp_proto_rndv_mtype_get_req_memh(req),
+                              req->send.rndv.mdesc->memh->mem_type,
```

Comment:
maybe reuse the mem_type from line 512 by caching it in a local var?

Reply:
ok